autoload/sy/fold.vim

```diff
@@ -88,11 +88,21 @@ endfunction
 
 " s:get_lines {{{1
 function! s:get_lines() abort
-  let signlist = sy#util#execute('sign place buffer='. b:sy.buffer)
-
   let lines = []
-  for line in split(signlist, '\n')[2:]
-    call insert(lines, matchlist(line, '\v^\s+line\=(\d+)')[1], 0)
+
+  let has_sign_func = has('patch-8.1.614')
+  if has_sign_func
+    let signlist = sign_getplaced(b:sy.buffer)[0].signs
+  else
+    let signlist = split(sy#util#execute('sign place buffer='. b:sy.buffer), '\n')[2:]
+  endif
```

Comment:
A better way to handle this is probably to define a wrapper in autoload/sy/util.vim.  Something like

```vim
if exists('*sign_getplaced')
  function! sy#util#get_signs(bufnr)
    return sign_getplaced(a:bufnr)
  endfunction
else
  function! sy#util#get_signs(bufnr)
    ... Parse ":sign place" output and return a similar data structure
  endfunction
endif
```

Then the places that call the wrapper code don't need to be complicated by the two approaches, and we only need to do the check once to define the function wrapper.